src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -242,18 +241,26 @@ def _get_docker_command():
             _, stderr = p.communicate()
         except OSError as inner:
             logger.debug("Could not run '%s' command. Exception: %s", docker_command, str(inner))
-            raise CLIError(docker_not_installed)
+            if is_diagnostics_context:
+                return None, DOCKER_COMMAND_ERROR
+            raise CLIError(DOCKER_COMMAND_ERROR.get_error_message())
         except CalledProcessError as inner:
             logger.debug("Could not run '%s' command. Exception: %s", docker_command, str(inner))
-            raise CLIError(docker_not_available)
+            if is_diagnostics_context:
+                return docker_command, DOCKER_DAEMON_ERROR
+            raise CLIError(DOCKER_DAEMON_ERROR.get_error_message())
     except CalledProcessError as e:
         logger.debug("Could not run '%s' command. Exception: %s", docker_command, str(e))
-        raise CLIError(docker_not_available)
+        if is_diagnostics_context:
+            return docker_command, DOCKER_DAEMON_ERROR
+        raise CLIError(DOCKER_DAEMON_ERROR.get_error_message())
 
     if stderr:
```

Comment:
`docker ps` will write a warning to stderr if `.docker/config.json` requires elevated access. For example, if there are secrets in that file. In that case, throwing an error is probably not necessary. We could ignore the warning and carry on.